internal/terraform/node_resource_plan_instance.go

```diff
@@ -547,6 +554,87 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 	return diags
 }
 
+// planActionTriggers plans all actions (potentially) triggered by this
+// resource instance change sequentially, and returns any diagnostics
+func (n *NodePlannableResourceInstance) planActionTriggers(ctx EvalContext, change *plans.ResourceInstanceChange) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+
+	if n.Config == nil || n.Config.Managed == nil || n.Config.Managed.ActionTriggers == nil {
+		return diags
+	}
+
+	for i, at := range n.Config.Managed.ActionTriggers {
+		if !eventIncludesAction(at.Events, change.Action) {
+			continue
+		}
+
+		// TODO: Deal with conditions
+		for j, actionRef := range at.Actions {
+			var actionAddr addrs.ActionInstance
+
+			ref, parseRefDiags := addrs.ParseRef(actionRef.Traversal)
+			diags = diags.Append(parseRefDiags)
+			if parseRefDiags.HasErrors() {
+				return diags
+			}
+
+			if a, ok := ref.Subject.(addrs.ActionInstance); ok {
+				actionAddr = a
+			} else if a, ok := ref.Subject.(addrs.Action); ok {
+				// Could be a reference to an action without an instance specified
+				// TODO: This is where we should auto-expand, for now we will just default to taking
```

Comment:
Is this still accurate, or just leftover from before you added the expansion code?

Reply:
Still valid, this is about this reference feature where when you reference an action that is expanded without am instance key you put all the referenced action instances into the list of actions to trigger. I pushed that to later, for now just doing the single instance references in this place. 

If you want me to add this feature already I can do that